materials/property.py

```diff
@@ -0,0 +1,135 @@
+"""classes for representing and querying properties of a material."""
+import numpy as np
+import scipy.interpolate
+
+class Property:
+    """A property of a material."""
+    def __init__(self, name, yaml_dict):
+        self.name = name
+        self.default_value = yaml_dict['default_value']
+        self.units = yaml_dict['units']
+        self.reference = yaml_dict['reference']
+
+    def query_value(self):
+        return self.default_value
+
+
+class StateDependentProperty(Property):
+    """A property of a material which depends on state (e.g. temperature)."""
+    def __init__(self, name, yaml_dict):
+        Property.__init__(self, name, yaml_dict)
```

Comment:
There's a few logical steps in this `__init__` -- I'd suggest breaking each into a named function less than 30 lines long.

Ideally, those functions wouldn't mutate the state of the class, and would instead consume arguments and return some newly created object the class should know about

Reply:
fixed by https://github.com/mvernacc/material-properties-interchange/pull/1/commits/f48f19c3a749890accddc1232ecf53498025fc36